code/game/objects/structures/icemoon/cave_entrance.dm

```diff
@@ -0,0 +1,178 @@
+
+/obj/structure/spawner/ice_moon/polarbear/clear_rock()
+	for(var/turf/F in RANGE_TURFS(1, src))
+		if(ismineralturf(F))
+			var/turf/closed/mineral/M = F
+			M.ScrapeAway(null, CHANGETURF_IGNORE_AIR)
+
+/obj/structure/spawner/ice_moon/demonic_portal
+	name = "demonic portal"
+	desc = "A portal that goes to another world, normal creatures couldn't survive there."
+	icon_state = "nether"
+	mob_types = list(/mob/living/simple_animal/hostile/asteroid/ice_demon)
+	light_range = 1
+	light_color = LIGHT_COLOR_RED
+
+/obj/structure/spawner/ice_moon/demonic_portal/clear_rock()
+	for(var/turf/F in RANGE_TURFS(3, src))
+		if(abs(src.x - F.x) + abs(src.y - F.y) > 5)
+			continue
+		if(ismineralturf(F))
+			var/turf/closed/mineral/M = F
+			M.ScrapeAway(null, CHANGETURF_IGNORE_AIR)
+
+/obj/structure/spawner/ice_moon/demonic_portal/destroy_effect()
+	new /obj/effect/collapsing_demonic_portal(loc)
+
+/obj/structure/spawner/ice_moon/demonic_portal/drop_loot()
+	return
+
+/obj/structure/spawner/ice_moon/demonic_portal/ice_whelp
+	mob_types = list(/mob/living/simple_animal/hostile/asteroid/ice_whelp)
+
+/obj/structure/spawner/ice_moon/demonic_portal/snowlegion
+	mob_types = list(/mob/living/simple_animal/hostile/asteroid/hivelord/legion/snow)
+
+/obj/effect/collapsing_demonic_portal
+	name = "collapsing demonic portal"
+	desc = "It's slowly fading!"
+	layer = TABLE_LAYER
+	icon = 'icons/mob/nest.dmi'
+	icon_state = "nether"
+	anchored = TRUE
+	density = TRUE
+
+/obj/effect/collapsing_demonic_portal/Initialize()
+	. = ..()
+	playsound(loc,'sound/effects/tendril_destroyed.ogg', 200, FALSE, 50, TRUE, TRUE)
+	visible_message("<span class='boldannounce'>[src] begins to collapse, cutting it off from this world!</span>")
+	animate(src, transform = matrix().Scale(0, 1), alpha = 50, time = 5 SECONDS)
+	addtimer(CALLBACK(src, .proc/collapse), 5 SECONDS)
+
+/obj/effect/collapsing_demonic_portal/proc/collapse()
+	visible_message("<span class='warning'>Something slips out of [src]!</span>")
+	var/loot = rand(1, 28)
+	switch(loot)
+		if(1)
+			new /obj/item/clothing/suit/space/hardsuit/cult(loc)
+		if(2)
+			new /obj/item/clothing/glasses/godeye(loc)
+		if(3)
+			new /obj/item/reagent_containers/glass/bottle/potion/flight(loc)
+		if(4)
+			new /obj/item/organ/heart/cursed/wizard(loc)
+		if(5)
+			new /obj/item/jacobs_ladder(loc)
+		if(6)
+			new /obj/item/rod_of_asclepius(loc)
+		if(7)
+			new /obj/item/warp_cube/red(loc)
+		if(8)
+			new /obj/item/wisp_lantern(loc)
+		if(9)
+			new /obj/item/immortality_talisman(loc)
+		if(10)
+			new /obj/item/book/granter/spell/summonitem(loc)
+		if(11)
+			new /obj/item/clothing/neck/necklace/memento_mori(loc)
+		if(12)
+			new /obj/item/borg/upgrade/modkit/lifesteal(loc)
+			new /obj/item/bedsheet/cult(loc)
+		if(13)
+			new /obj/item/disk/design_disk/modkit_disc/mob_and_turf_aoe(loc)
+		if(14)
+			new /obj/item/disk/design_disk/modkit_disc/bounty(loc)
+		if(15)
+			new /obj/item/ship_in_a_bottle(loc)
```

Comment:
does the ice moon have ice rivers or something?

if not, consider replacing this loot drop with something else

Reply:
plasma rivers are on the bottom z level